content/blog/registry-writing-i2c-component/index.md

```diff
@@ -0,0 +1,742 @@
+
+On this article, we will focus on the GitHub Action procedure.
+
+#### GitHub Actions
+
+If you are not familiar with GitHub Actions, you can read the [official documentation](https://github.com/features/actions).
+
+In sum, GitHub Action is a continuous integration and continuous delivery (CI/CD) platform that automates workflows right on in your GitHub repository. You can use GitHub Actions in a vast situations and you can automate for example the component publishing process.
+
+#### Workflow
+
+To use the actions, we need to create a folder on the root directory of your component repository, named `.github` (do not forget the dot).
+
+Inside this folder, create another folder named `workflows`. On this folder, two workflow files will be created.
+
+- `build_examples.yml`: This workflow will build the component example.
+- `upload_components.yml`: This workflow will upload the component to the registry.
+
+The folder structure will be:
+
+```text
+.
+.github
+└── workflows
+    └── build_examples.yml
+    └── upload_components.yml
+└── shtc3
+```
+
+**build_examples.yml**
+
+The build workflow will compile the example with the ESP-IDF version specified in the `idf_ver` and will run when you push to the main branch or create a PR (pull request).
+
+This workflow is important to ensure that the code can be built successfully, preventing any potential issues before uploading the component to the registry.
+
+```yaml
+name: 'build'
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        idf_ver: ["release-v5.3", "latest"]
+    runs-on: ubuntu-22.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4
+    - name: Install idf-build-apps
+      shell: bash
+      run: |
+        . ${IDF_PATH}/export.sh
+        python3 -m pip install idf-build-apps==2.4.3
+    - name: Build examples
+      shell: bash
+      run: |
+        . ${IDF_PATH}/export.sh
+        idf-build-apps build
+```
+
+**upload_components.yml**
+
+To publish the component, the workflow `upload_component` will process the component by using the [upload-components-ci-action](https://github.com/espressif/upload-components-ci-action).
+
+This workflow will run only when the component is pushed to the main branch. This avoids publishing the component before merging the branch to the main.
+
+An important note is that the action will only publish the component if there is no component published with the same version. This means that you need to change the version in the manifest file before running this workflow.
+
+```yaml
+name: Push components to Espressif Component Service
+
+on:
+    push:
+      branches:
+        - main
+
+jobs:
+  upload_components:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+
+      - name: Upload components to component service

```

Comment:
Consider adding CI tests before pushing the component to registry: 
https://github.com/georgik/esp32-sdl3-example/blob/main/.github/workflows/test.yml

Reply:
I prefer to write something later.